common/src/main/java/com/wynntils/gui/render/buffered/BufferedFontRenderer.java

```diff
@@ -355,4 +357,110 @@ public void renderText(
                     shadow);
         }
     }
+
+    public void renderTextsWithAlignment(
+            PoseStack poseStack,
+            MultiBufferSource.BufferSource bufferSource,
+            float x,
+            float y,
+            List<TextRenderTask> toRender,
+            float width,
+            float height,
+            HorizontalAlignment horizontalAlignment,
+            VerticalAlignment verticalAlignment) {
+        float renderX =
+                switch (horizontalAlignment) {
+                    case Left -> x;
+                    case Center -> x + width / 2;
+                    case Right -> x + width;
+                };
+
+        float renderY =
+                switch (verticalAlignment) {
+                    case Top -> y;
+                    case Middle -> y + (height - FontRenderer.getInstance().calculateRenderHeight(toRender)) / 2;
+                    case Bottom -> y + (height - FontRenderer.getInstance().calculateRenderHeight(toRender));
+                };
+
+        renderTexts(poseStack, bufferSource, renderX, renderY, toRender);
+    }
+
+    public void renderTexts(
+            PoseStack poseStack,
+            MultiBufferSource.BufferSource bufferSource,
+            float x,
+            float y,
+            List<TextRenderTask> lines) {
+        float currentY = y;
+        for (TextRenderTask line : lines) {
+            renderText(poseStack, bufferSource, x, currentY, line);
+            // If we ask Mojang code the line height of an empty line we get 0 back so replace with space
```

Comment:
I think this is copied from the other renderTexts(). Can you extract this into a common `getLineHeight()` or something?

Reply:
There are other height methods which are only a bit different so it would be a bit confusing imo

Reply:
Ok, leave it then. But I think we need to revisit this code and clean it up a bit later on. I always feel like there is a ton of "renderText" which calls other "renderText" ad nauseum until we end up in a "renderText" that actually does something. :-)

Reply:
Yes, agreed. Ideally, we would get rid of the regular rendering. I think. Not so sure then.